python/tvm/relay/frontend/onnx.py

```diff
@@ -898,7 +898,7 @@ def __init__(self, shape, dtype):
         self._renames = {}
         self._num_input = 0
         self._num_param = 0
-        self._shape = shape
+        self._shape = shape if shape else {}
```

Comment:
Is this for error reporting reasons?

Reply:
Yes, otherwise pulling value from None is ambiguous error.

Reply:
Cool, just ping me when CI is green and I will merge. Any thoughts on being robust to non-topo sorted models?

Reply:
Sure. 
For non-topo onnx models, I guess it's doable. However I am not conviced so far to bring in a sorting function purely for what should supposed to be fixed by onnx/keras-onnx

Reply:
Yeah I agree, we shouldn't be fixing others bugs, is there some where that states it must be topo-sorted?

Reply:
here's the claim: https://github.com/onnx/onnx/blob/master/docs/IR.md#graphs

Reply:
Awesome, I think we should add a note in the docs as a warning, what do you think?

Reply:
I also saw there are some onnx utilities which check the graph, do you know if these check topo. ordering?

Reply:
actually onnx checker is a good idea